store/elasticsearch/type_repository.go

```diff
@@ -39,32 +39,32 @@ type TypeRepository struct {
 	cli *elasticsearch.Client
 }
 
-func (repo *TypeRepository) CreateOrReplace(ctx context.Context, recordType record.Type) error {
-	if isReservedName(recordType.Name) {
-		return record.ErrReservedTypeName{TypeName: recordType.Name}
+func (repo *TypeRepository) CreateOrReplace(ctx context.Context, recordTypeName record.TypeName) error {
```

Comment:
We may not need this function anymore if we are going with type as an enum.

Reply:
In fact, is it possible to remove TypeRepository? Or do we still need it for something else?

Reply:
We need this for migration.

Reply:
Oh yeah true!
Would it make sense if we move `record.TypeRepository.CreateOrReplace` to `discovery.Migrate(typeName)`?
wdyt?

Reply:
hmm I think better to separate the responsibility to repository layer

Reply:
Oh true forgot about it 🙏.  I mean `store/elasticsearch.Migrate(typeName)`.

Reply:
Hmm yes that might be doable. Will try it.

Reply:
@StewartJingga I think better to just keep it in typeRepository, because other functions need that to test